src/components/MessageInput/hooks/useTimer.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+
+export const useTimer = ({ startFrom }: { startFrom: number }) => {
```

Comment:
Can you check if this can be reused in `CooldownTimer` component? Not necessarily in this PR :)

Reply:
This code has been extract from CooldownTimer and imported there :)